### ISSUE TITLE
Missing documentation/semantics in `Model.call()`/`__call__()` (`mask` and relationship with `__call__()`

### ISSUE THREAD

**Issue by lebigot** [keras-team/keras#15185]
**URL(s) with the issue:**.

https://www.tensorflow.org/api_docs/python/tf/keras/Model#call

**Description of issue (what needs to be changed):**.

1. The documentation/semantics for `mask` in `Model.call()`/`__call__()` is missing. It would be useful to at least point people to the relevant information on `mask` (if it exists somewhere) or to spell it out.

    Currently, users have no idea of what `mask` represents by simply reading the documentation for `call()`. Pointing once and for all to the relevant documentation (or writing it) would save a lot of collective time.

2. Is the semantics / parameters / etc. of `__call__()` the same as those of `call()`? It would be useful to spell this out explicitly (it seems to be the case), along with pointing to the semantics of `__call__()`.

**Parameters defined**.

Are all parameters defined and formatted correctly? No.

Thanks!

**Comment by jvishnuvardhan** [comment]
@lebigot Thanks for the opening the issue. 

Regarding (1), are you looking for more detailed info on `mask`. Please check [this page](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Masking) and the [guide](https://www.tensorflow.org/guide/keras/masking_and_padding) on masking.

Regarding (2), i think some internal details are intentionally avoided in the docs to simplify docs for most of the users. Please call a model as model(x). [This](https://stackoverflow.com/questions/57103604/why-keras-use-call-instead-of-call) may be helpful. Thanks!

**Comment by lebigot** [comment]
While (1) is interesting, my point is that a lot of collective time would be saved if what you write for (1) *were in the documentation*. This is what I suggest. 

As for (2), my point was that what `call()` does and returns is only implicit: "Calls the model on new inputs" is in fact less explicit than "Returns the outputs of the model for new inputs, as tensors." (like `__call__()` does), which is what I would suggest could instead usefully be in the documentation. Explicit beats implicit. :) What do you think?
